platform/src/main/java/org/stellar/anchor/platform/controller/Sep24Controller.java

```diff
@@ -0,0 +1,208 @@
+      consumes = {MediaType.ALL_VALUE},
+      method = {RequestMethod.GET})
+  public GetTransactionsResponse getTransactions(
+      HttpServletRequest request,
+      @RequestParam(value = "asset_code") String assetCode,
+      @RequestParam(required = false, value = "kind") String kind,
+      @RequestParam(required = false, value = "limit") Integer limit,
+      @RequestParam(required = false, value = "paging_id") String pagingId,
+      @RequestParam(required = false, value = "no_older_than") String noOlderThan)
+      throws MalformedURLException, URISyntaxException, SepException {
+    debugF(
+        "/transactions asset_code={} kind={} limit={} no_older_than={} paging_id={}",
+        assetCode,
+        kind,
+        limit,
+        noOlderThan,
+        pagingId);
+    GetTransactionsRequest gtr =
+        GetTransactionsRequest.of(assetCode, kind, limit, noOlderThan, pagingId);
+    return getTransactions(request, gtr);
+  }
+
+  @CrossOrigin(origins = "*")
+  @RequestMapping(
+      value = "/transaction",
+      consumes = {MediaType.APPLICATION_JSON_VALUE},
+      method = {RequestMethod.GET})
+  public GetTransactionResponse getTransaction(
+      HttpServletRequest request, @RequestBody GetTransactionRequest tr)
+      throws SepException, IOException, URISyntaxException {
+    debug("/transaction", tr);
+    JwtToken token = getSep10Token(request);
+
+    return sep24Service.findTransaction(token, tr);
+  }
+
+  @CrossOrigin(origins = "*")
+  @RequestMapping(
+      value = "/transaction",
+      consumes = {MediaType.ALL_VALUE},
+      method = {RequestMethod.GET})
+  public GetTransactionResponse getTransaction(
+      HttpServletRequest request,
+      @RequestParam(required = false, value = "id") String id,
+      @RequestParam(required = false, value = "external_transaction_id")
+          String externalTransactionId,
+      @RequestParam(required = false, value = "stellar_transaction_id") String stellarTransactionId)
+      throws SepException, IOException, URISyntaxException {
+    debugF(
+        "/transaction id={} external_transaction_id={} stellar_transaction_id={}",
+        id,
+        externalTransactionId,
+        stellarTransactionId);
+    GetTransactionRequest tr =
+        new GetTransactionRequest(id, stellarTransactionId, externalTransactionId);
+    return getTransaction(request, tr);
+  }
+
+  String getFullRequestUrl(HttpServletRequest request) {
+    if (request.getQueryString() != null) {
+      return request.getRequestURL() + "?" + request.getQueryString();
+    }
+
+    return request.getRequestURL().toString();
+  }
+
+  JwtToken getSep10Token(HttpServletRequest request) throws SepValidationException {
+    JwtToken token = (JwtToken) request.getAttribute(JWT_TOKEN);
+    if (token == null) {
+      throw new SepValidationException(
+          "missing sep10 jwt token. Make sure the sep10 filter is invoked before the controller");
+    }
+    return token;
+  }
```

Comment:
👀  You don't need to implement this method here since it's already implemented in Sep10Helper:
https://github.com/stellar/java-stellar-anchor-sdk/blob/657e97cd623da09160392b3ed2044a554ecfcf82/platform/src/main/java/org/stellar/anchor/platform/controller/Sep10Helper.java#L9-L18

Reply:
I don't know how you can catch this! 😲 

Fixed.

Reply:
I remembered we already have something in place for that 😄